flask_core/app.py

```diff
@@ -47,8 +76,11 @@ def create_app(config=None):
 
     # Register all our middleware
     app.wsgi_app = Handler(app.wsgi_app)
+    # Set up magic flag generator (ty closure)
+    app.gen_flag = types.MethodType(gen_flag, app)
+    app.check_flag = types.MethodType(check_flag, app)
 
+    app.after_request(types.MethodType(grep_flag, app))
```

Comment:
Attach the function directly without `MethodType`, `from flask import current_app`